src/bosh-director/lib/bosh/director/jobs/scheduled_tasks_cleanup.rb

```diff
@@ -15,6 +17,34 @@ def initialize(_param = {})
         @task_remover = Bosh::Director::Api::TaskRemover.new(Config.max_tasks)
       end
 
+      def update_orphaned_tasks_with_state_error(result)
+        actual_delayed_job_scheduled_task_ids = []
+        Delayed::Worker.backend = :sequel
+        Delayed::Job.all.select do |job|
+          actual_delayed_job_scheduled_task_ids << YAML.safe_load(job.handler)['task_id']
+        end
+
+        errored_tasks = []
+        Bosh::Director::Models::Task.select.where('state': %w[queued processing]).each do |task|
+          next if actual_delayed_job_scheduled_task_ids.include?(task.id)
+
+          # Newly created tasks are first created in the tasks db, then in the delayed jobs db.
+          # So if a task is being created while we get the list of all jobs in the delayed jobs table,
+          # we may end up missing an item.
+          # To avoid cleaning up tasks where the delayed job item doesn't exist yet, we only clean tasks
+          # if the timestamp older than 5 minutes.
+          next unless (DateTime.now.to_time.to_i - task.timestamp.to_i) >= 300
+
+          errored_tasks << task.id
+          task.state = 'error'
+          task.save
```

Comment:
I was wondering whether we should do something here in case the `task.save` fails like we do [here](https://github.com/cloudfoundry/bosh/pull/2413/files#diff-0628a0099bcf26e0203cbe9d841fc5e72389d8926054b88d96f1112ee111ad17R71). Failing the operation doesn't make much sense because the task is anyway removed and we will leave also the rest of the tasks orphaned. Maybe we can log this for debug reasons.

Reply:
done